File: dp-core/src/main/java/com/dataplatform/org/entity/Post.java
package com.dataplatform.org.entity;

import lombok.Data;
import java.time.LocalDateTime;

/**
 * 岗位实体
 */
@Data
public class Post {

    private Long id;

    /** 岗位编码（唯一） */
    private String postCode;

    /** 岗位名称 */
    private String postName;

    /** 排序号 */
    private Integer sortOrder;

    /** 状态: 0-禁用 1-启用 */
    private Integer status;

    /** 备注 */
    private String remark;

    /** 创建时间 */
    private LocalDateTime createTime;

    /** 更新时间 */
    private LocalDateTime updateTime;
}

File: dp-ui/src/api/org/post.ts
import request from '@/api/request'

// ==================== 类型定义 ====================

/** 岗位 */
export interface Post {
  id?: number
  postCode: string
  postName: string
  sortOrder?: number
  status?: number
  remark?: string
  createTime?: string
  updateTime?: string
}

// ==================== 岗位接口 ====================

/** 查询岗位列表（支持按名称和状态筛选，按 sortOrder 升序） */
export const listPosts = (params?: { postName?: string; status?: number | null }) => {
  return request.get('/posts', { params })
}

/** 查询岗位详情 */
export const getPostById = (id: number) => {
  return request.get(`/posts/${id}`)
}

/** 创建岗位 */
export const createPost = (data: Post) => {
  return request.post('/posts', data)
}

/** 更新岗位 */
export const updatePost = (id: number, data: Post) => {
  return request.put(`/posts/${id}`, data)
}

/** 删除岗位（检查关联用户） */
export const deletePost = (id: number) => {
  return request.delete(`/posts/${id}`)
}

/** 查询岗位下的用户列表 */
export const getPostUsers = (id: number) => {
  return request.get(`/posts/${id}/users`)
}
